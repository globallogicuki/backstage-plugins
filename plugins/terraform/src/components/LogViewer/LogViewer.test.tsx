import React from 'react';
import useAsync from 'react-use/lib/useAsync';
import { render, screen } from '@testing-library/react';
import { LogViewer } from './LogViewer';

jest.mock('react-use/lib/useAsync');

jest.mock('@backstage/core-components', () => {
  const originalModule = jest.requireActual('@backstage/core-components');

  const MockLogViewer = ({ text }: { text: string }) => (
    <div>{`Mock TerraformRuns: ${text}`}</div>
  );

  return {
    __esModule: true,
    LogViewer: MockLogViewer,
    Progress: originalModule.Progress,
  };
});

describe('LogViewer', () => {
  beforeEach(() => {
    (useAsync as jest.Mock).mockReturnValue({
      value: undefined,
      loading: true,
      error: undefined,
    });
  });

  afterEach(() => {
    (useAsync as jest.Mock).mockRestore();
  });

  it('renders when loading', async () => {
    const { container } = render(<LogViewer txtUrl="someurl.txt" />);

    expect(container).toMatchSnapshot();
  });

  it('renders when error', async () => {
    (useAsync as jest.Mock).mockReturnValue({
      value: undefined,
      loading: false,
      error: new Error('Some fake error.'),
    });

    render(<LogViewer txtUrl="someurl.txt" />);

    const error = await screen.findByText(/Some fake error./i);

    expect(error).toBeInTheDocument();
  });

  it('renders when successful', async () => {
    (useAsync as jest.Mock).mockReturnValue({
      value: 'this is some text to be shown',
      loading: false,
      error: undefined,
    });

    render(<LogViewer txtUrl="someurl.txt" />);

    const text = await screen.findByText(/this is some text to be shown/i);

    expect(text).toBeInTheDocument();
  });
});
