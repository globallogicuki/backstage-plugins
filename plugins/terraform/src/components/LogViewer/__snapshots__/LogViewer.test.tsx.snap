// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LogViewer renders when loading 1`] = `
<div>
  <div
    class="MuiBox-root MuiBox-root-1"
    data-testid="progress"
  />
</div>
`;
